onnxruntime/core/session/environment.cc

```diff
@@ -746,6 +746,28 @@ std::vector<const OrtHardwareDevice*> SortDevicesByType() {
 
   return sorted_devices;
 }
+
+bool AreVirtualDevicesAllowed(std::string_view lib_registration_name) {
+  constexpr std::string_view suffix{".virtual"};
+
+  return lib_registration_name.size() >= suffix.size() &&
+         lib_registration_name.compare(lib_registration_name.size() - suffix.size(),
+                                       suffix.size(), suffix) == 0;
+}
+
+Status SetEpFactoryEnvironmentOptions(OrtEpFactory& factory, std::string_view lib_registration_name) {
+  if (factory.SetEnvironmentOptions == nullptr) {
```

Comment:
needs version check as well. could be random bits if the EP was built with ORT 1.23

Reply:
Thanks!